### PR TITLE
Improve openssl skipifs and adjust Crypto package module

### PR DESCRIPTION
**Pull request by mppf**
* Improves skipifs for tests using openssl to skip if libcrypto isn't installed and to work with openssl version 1.1.1k
* Adjusts CryptoBuffer.init(string) to work correctly with Unicode and added a bytes version
* Deprecated Blowfish support in Crypto because
  * it does not work as written in OpenSSL 3
  * Blowfish is considered legacy in OpenSSL 3

`start_test test/studies/dedup/dedup-extern.chpl  test/studies/dedup/dedup-externblock.chpl  test/library/packages/Crypto --respect-skipifs` passes:

- [x] with no libcrypto installed, everything is skipped
- [x] with OpenSSL 3, skipping the Blowfish tests
- [x] with OpenSSL 1.1.k

Reviewed by @ronawho - thanks!

### CODE REVIEW

File: test/library/packages/Crypto.skipif [hunk 1]
```diff
@@ -12,14 +12,14 @@ Ok = False
...
-    v = subprocess.check_output(['pkg-config', '--modversion', 'libcrypto'])
-    v = v.rstrip()
-    v = str(v, encoding='utf-8', errors='surrogateescape')
-
-    if parse_version(v) >= parse_version('1.1'):
-        # OK, version is compatible.
-        Ok = True
-
+    sub = subprocess.run(['pkg-config', '--modversion', 'libcrypto'],
+                         capture_output=True)
```
Comment (ronawho) [inline test/library/packages/Crypto.skipif:16]: Based on the code here, you probably saw that `--modversion` fails if the package isn't installed and the `parse_version` isn't very good for pkg-config output. `pkg-config` actually has built-in version comparisons, so I think we improve the portability of this and simplify it with:

```python
    sub = subprocess.run(['pkg-config', '--exists', 'libcrypto >= 1.1'])
    if sub.returncode == 0:
        Ok = True
```

Comment (ronawho) [reply test/library/packages/Crypto.skipif:16]: And just some sample output on a machine without libcrypto and with it:

Without:
```sh
chaplogin01:~> pkg-config --modversion libcrypto
No package 'libcrypto' found
chaplogin01:~> pkg-config --exists 'libcrypto >= 1.1'
chaplogin01:~> echo $?
1
```

With:
```sh
chapcs11: ~ $ pkg-config --modversion libcrypto
1.1.1k
chapcs11: ~ $ pkg-config --exists 'libcrypto >= 1.1'
chapcs11: ~ $ echo $?
0
```

With, but newer version:

```sh
chapcs11: ~ $ pkg-config --exists 'libcrypto >= 1.2'
chapcs11: ~ $ echo $?
1
```


```

File: test/studies/dedup/dedup-extern.skipif [hunk 2]
```diff
@@ -1 +1,18 @@
-CHPL_TARGET_PLATFORM != linux64
+#!/usr/bin/env python3
+
+import os
+import subprocess
+from pkg_resources import parse_version
```
Comment (ronawho) [inline test/studies/dedup/dedup-extern.skipif:5]: Can you remove the `parse_version` import from the skipifs?

File: test/library/packages/Crypto.skipif (hunk 1 above)
Comment (mppf) [reply test/library/packages/Crypto.skipif]: Thanks for the suggestion. I think it's a great idea & I've made the change.

**Comment by ronawho** [comment]
Thanks!

**Pull request merged**